vyper/compiler.py

```diff
@@ -47,6 +50,7 @@
 
 def __compile(code, interface_codes=None, *args, **kwargs):
     ast = parse_to_ast(code)
+    validate_semantics(ast, interface_codes)
```

Comment:
So, validate does constant folding and other node-modifying interactions? Therefore, this is a stateful operation?

I'd like to know what it would take to make this a stateless operation